### DISCUSSION TITLE
SQLModel.validate: copy_on_model_validation broken by Pydantic 1.9

### DISCUSSION THREAD

**Discussion by jlost** [fastapi/sqlmodel#676] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from sqlmodel import Field, Session, SQLModel, create_engine


class Hero(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    secret_name: str
    age: Optional[int] = None

    class Config:
        copy_on_model_validation = "none"


hero_1 = Hero(name="Deadpond", secret_name="Dive Wilson")

engine = create_engine("sqlite:///database.db")


SQLModel.metadata.create_all(engine)

with Session(engine) as session:
    session.add(hero_1)
    session.commit()
    session.refresh(hero_1)
    print(hex(id(hero_1)))
    validated = Hero.validate(hero_1)
    print(hex(id(validated)))




pyproject.toml:

[tool.poetry]
name = "sqlmodel-demo"
version = "0.1.0"
description = ""
authors = ["jlost"]
readme = "README.md"
packages = [{include = "sqlmodel_demo"}]

[tool.poetry.dependencies]
python = "^3.11"
sqlmodel = "^0.0.9"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

### Description

Pydantic 1.9 added `none`/`shallow`/`deep` as options for `pydantic.config.BaseConfig.copy_on_model_validation`. Before that, it was a bool.
https://github.com/pydantic/pydantic/commit/47362e33b630ed60e27ec95d798ca013f602b4e4

SQLModel checks it as if it were a boolean in main.py:
https://github.com/tiangolo/sqlmodel/blob/d05c3ee495bb81d2915b38052ac5f1909da60e37/sqlmodel/main.py#L596
Because `copy_on_model_validation` is a str now, this will always return True.

Ideas for solutions:
* `return value.copy() if cls.__config__.copy_on_model_validation != "none" else value`
  * Not sure whether more would need to be done to support the shallow/deep distinction.
* ~Pin Pydantic dependency to <1.9~
  * ~Looks to me like Pydantic broke compatibility without incrementing the major version, but this would fix it, and it looks like 1.8.2 is still compatible with FastAPI.~
  * 0.0.10 bumps minimum Pydantic to 1.9, so this is no longer an option.
* Workaround: Set `copy_on_model_validation` to None even though that isn't a valid value per Pydantic's type hints
  * Might have side effects, probably not a good idea.

Thoughts? More than happy to create a PR you think this needs fixing!

### Operating System

Linux

### Operating System Details

Fedora 38

### SQLModel Version

0.0.9, 0.0.10, 0.0.12

### Python Version

3.11

**Comment by jlost** [comment]
I see 0.0.10 bumps minimum Pydantic to 1.9, so I guess pinning Pydantic < 1.9 is no longer an option.

**Comment by jlost** [comment]
Problem remains in 0.0.12.

**Comment by YuriiMotov** [comment]
Works well in current version (SQLModel 0.0.24, Pydantic 1.10.22)
